packages/gestalt/src/Video/VideoPlayhead.js

```diff
@@ -0,0 +1,109 @@
+// @flow
+
+import * as React from 'react';
+import Box from '../Box/Box';
+import PropTypes from 'prop-types';
+import styles from './Video.css';
+
+type Props = {|
+  currentTime: number,
+  duration: number,
+  seek: (time: number) => void,
+|};
+
+type State = {|
+  seeking: boolean,
+|};
+
+export default class VideoPlayhead extends React.PureComponent<Props, State> {
+  state = {
+    seeking: false,
+  };
+
+  setPlayheadRef = (ref: ?HTMLDivElement) => {
+    this.playhead = ref;
+  };
+
+  seek = (clientX: number) => {
+    if (this.playhead) {
+      const { duration, seek } = this.props;
+      const { left, width } = this.playhead.getBoundingClientRect();
+      const percent = Math.max(0, Math.min((clientX - left) / width, 1));
+      const newTime = percent * duration;
+      seek(newTime);
+    }
+  };
+
+  handleMouseDown = (event: SyntheticMouseEvent<HTMLDivElement>) => {
+    this.setState({ seeking: true });
+    this.seek(event.clientX);
+  };
+
+  handleMouseMove = (event: SyntheticMouseEvent<HTMLDivElement>) => {
+    const { seeking } = this.state;
+    if (seeking) {
+      this.seek(event.clientX);
+    }
+  };
+
+  handleMouseUp = () => {
+    this.setState({ seeking: false });
+  };
+
+  playhead: ?HTMLDivElement;
+
+  render() {
+    const { currentTime, duration } = this.props;
+    const width = `${Math.floor(currentTime * 10000 / duration) / 100}%`;
+    return (
+      <Box
+        position="relative"
+        display="flex"
+        flex="grow"
+        alignItems="center"
+        height={16}
+      >
+        <div
+          role="presentation"
```

Comment:
@pfarejowicz this piece is completely inaccessible - could we use the `progressbar` role or the `<progress>` element?

Also supplying the right aria labels: e.g. `aria-valuenow="20" aria-valuemin="0" aria-valuemax="100"`

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_progressbar_role